File: frontend/src/api/planner.ts
import { request } from "./client";

export interface PlanPayload {
  destination: string;
  days: number;
  travel_style: string;
  travelers: string;
  preferences: string[];
  constraints: string[];
  budget_level: string;
  model?: string;
}

export interface PlanResult {
  summary: string;
  itinerary: Array<{
    day: number;
    period: string;
    activity: string;
    reason: string;
  }>;
  culture_tips: string[];
  risk_alerts: string[];
  signal_basis: string[];
}

export interface ModelOption {
  value: string;
  label: string;
  desc: string;
}

export function createPlan(payload: PlanPayload): Promise<PlanResult> {
  return request<PlanResult>("/plan", {
    method: "POST",
    body: JSON.stringify(payload),
  });
}

export function fetchModels(): Promise<ModelOption[]> {
  return request<ModelOption[]>("/models");
}

File: backend/app/schemas/planner.py
from typing import List

from pydantic import BaseModel, Field


class PlanRequest(BaseModel):
    destination: str = Field(..., description="目的地，例如汕头/潮州/揭阳")
    days: int = Field(default=2, ge=1, le=7)
    travel_style: str = Field(default="轻松")
    travelers: str = Field(default="朋友")
    preferences: List[str] = Field(default_factory=list)
    constraints: List[str] = Field(default_factory=list)
    budget_level: str = Field(default="中等")
    model: str | None = Field(default=None, description="可选模型ID，留空则使用默认模型")


class ItineraryItem(BaseModel):
    day: int
    period: str
    activity: str
    reason: str


class PlanResponse(BaseModel):
    summary: str
    itinerary: List[ItineraryItem]
    culture_tips: List[str]
    risk_alerts: List[str]
    signal_basis: List[str]


class ModelOption(BaseModel):
    value: str
    label: str
    desc: str
